### PR TITLE
Fix #960 and #927: Oddness with git and non-root

### LINKED ISSUE #960: project.git=true inside git repo causes FNFEs
Scalafmt fails to correctly detect path for files if project is under git and `project.git=true`

### Workspace
```
[/tmp/scalafmt-git-test]$ tree -a
.
├── bin
│   └── scalafmt
├── .git
│   └── <standard git stuff, skipping>
└── scala-sources
    └── build.sbt

[/tmp/scalafmt-git-test]$ bin/scalafmt --version
scalafmt 1.0.0-RC1

[/tmp/scalafmt-git-test]$ git --version
git version 2.13.0
````

### `project.git=false` → scalafmt works
```
[/tmp/scalafmt-git-test/scala-sources]$ ../bin/scalafmt --config-str 'project.git=false' --test --debug
Working directory: /tmp/scalafmt-git-test/scala-sources
Formatting files: List(FileContents(/tmp/scalafmt-git-test/scala-sources/build.sbt))
Exception in thread "main" org.scalafmt.Error$MisformattedFile: /tmp/scalafmt-git-test/scala-sources/build.sbt is mis-formatted. 
	...
```

### `project.git=true` → scalafmt fails
```
[/tmp/scalafmt-git-test/scala-sources]$ ../bin/scalafmt --config-str 'project.git=true' --test --debug
Working directory: /tmp/scalafmt-git-test/scala-sources
Formatting files: List(FileContents(/tmp/scalafmt-git-test/build.sbt))
Exception in thread "main" java.io.FileNotFoundException: /tmp/scalafmt-git-test/build.sbt (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at org.scalafmt.util.FileOps$.readFile(FileOps.scala:63)
	at org.scalafmt.util.FileOps$.readFile(FileOps.scala:51)
	at org.scalafmt.cli.InputMethod$FileContents.readInput(InputMethod.scala:40)
	at org.scalafmt.cli.Cli$.handleFile(Cli.scala:112)
	at org.scalafmt.cli.Cli$.$anonfun$run$1(Cli.scala:175)
	at org.scalafmt.cli.Cli$.$anonfun$run$1$adapted(Cli.scala:173)
	at scala.collection.Iterator.foreach(Iterator.scala:929)
	...
```
For some reason, drops `scala-sources` from path

### Reproducible setup

```
[/tmp]$ mkdir -p scalafmt-git-test/{bin,scala-sources}

[/tmp]$ cd scalafmt-git-test

[/tmp/scalafmt-git-test]$ coursier bootstrap com.geirsson:scalafmt-cli_2.12:1.0.0-RC1 -o bin/scalafmt -f --main org.scalafmt.cli.Cli

[/tmp/scalafmt-git-test]$ cat > scala-sources/build.sbt
lazy val root   =    (project in file("."))

[/tmp/scalafmt-git-test]$ git init && git add . && git commit -am "Init"
Initialized empty Git repository in /tmp/scalafmt-git-test/.git/
[master (root-commit) e1b9d2a] Init
 2 files changed, 1 insertion(+)
 create mode 100755 bin/scalafmt
 create mode 100644 scala-sources/build.sbt
```

### PR DESCRIPTION
**Pull request by pjrt**
Fixes an issue where git commands would act oddly if `scalafmt` is called
from where NOT at the git root.

### CODE REVIEW

File: scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala [hunk 1]
```diff
@@ -45,36 +45,41 @@ class GitOpsImpl(workingDirectory: AbsoluteFile) extends GitOps {
             "-r",
             "HEAD",
             "--name-only",
+            "--full-name",
```
Comment (pjrt) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala]: `--full-name` gives back the name of the files "from the git root". This allows us to just append it to the `rootDir`.

File: scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala [hunk 2]
```diff
@@ -45,36 +45,41 @@ class GitOpsImpl(workingDirectory: AbsoluteFile) extends GitOps {
...
-        "--diff-filter=d",
-        branch
-      )).lines.map { x =>
-      AbsoluteFile.fromFile(new File(x), workingDirectory)
-    }.toSeq
+    val cmd = Seq(
+      "git",
+      "diff",
+      "--name-only",
+      "--diff-filter=d",
```
Comment (pjrt) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala]: `git-diff` does not have a `--full-name`, but no matter, it already returns files from the root of git.

File: scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala (hunk 1 above)
Comment (olafurpg) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala]: Good catch, I didn't know about --full-name 👍

File: scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala [hunk 3]
```diff
@@ -45,36 +45,41 @@ class GitOpsImpl(workingDirectory: AbsoluteFile) extends GitOps {
...
-      val cmd = Seq(
-        "git",
-        "rev-parse",
-        "--show-toplevel"
-      )
-      val result = AbsoluteFile.fromFile(new File(exec(cmd)), workingDirectory)
-      require(result.jfile.isDirectory)
-      result
-    }.toOption
+  override def rootDir: Option[AbsoluteFile] = {
```
Comment (olafurpg) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala]: This is cleaner, I've definitely changed my coding style over the past months

**Pull request merged**